aiokafka/admin/client.py

```diff
@@ -605,3 +612,66 @@ async def list_consumer_group_offsets(
                 offset_plus_meta = OffsetAndMetadata(offset, metadata)
                 response_dict[tp] = offset_plus_meta
         return response_dict
+
+    async def delete_records(
+        self,
+        records_to_delete: Dict[TopicPartition, RecordsToDelete],
+        timeout_ms: Optional[int] = None,
+    ) -> Dict[TopicPartition, int]:
+        """Delete records from partitions.
+
+        :param records_to_delete: A map of RecordsToDelete for each TopicPartition
+        :param timeout_ms: Milliseconds to wait for the deletion to complete.
+        :return: Appropriate version of DeleteRecordsResponse class.
+        """
+        version = self._matching_api_version(DeleteRecordsRequest)
+
+        if self._version_info[MetadataRequest[0].API_KEY] < (0, 10):
+            metadata_request = MetadataRequest[0]([])
+        else:
+            metadata_request = MetadataRequest[1](None)
+
+        metadata = await self._send_request(metadata_request)
```

Comment:
```suggestion
        metadata = await self._get_cluster_metadata(topics=None)
```

Reply:
With higher metadata protocol version `self._client.cluster.update_metadata(metadata)` doesn't work in the current version

```
            if error_type is Errors.NoError:
                _new_partitions[topic] = {}
>               for p_error, partition, leader, replicas, isr in partitions:
E               ValueError: too many values to unpack (expected 5)
```

Should I rather try to fix `update_metadata` or use the existing snippet to restrict ourselves to v0 or v1 ?

Reply:
I did the fix (that should be backward compatible)

Reply:
I think it would be better to fix it. As I understand, it could be as simple as
```python
for p_error, partition, leader, replicas, isr, *_ in partitions:
```
\+ a comment with explanation. Or do you see some reason against this?

Reply:
I just did the same fix, but without the comment, let me add one